### DISCUSSION TITLE
useSession Hook isn't returning accessToken

### DISCUSSION THREAD

**Discussion by harshithpabbati** [nextauthjs/next-auth#1328] [Help] [answered]
**Describe the bug**

useSession hook isn't returning accessToken, but it should return according to the [example session object](https://next-auth.js.org/getting-started/client#example-session-object).

**Steps to reproduce**
Console log session object from the following [example](https://github.com/nextauthjs/next-auth-example)

**Expected behavior**
It shoud return user, accessToken and expires. But only returns user, expires.

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by balazsorban44** [comment, marked as answer]
It does not do that by default.

See the TIP section right under the link you posted:
> The session data returned to the client does not contain sensitive information such as the Session Token or OAuth tokens. It contains a minimal payload that includes enough data needed to display information on a page about the user who is signed in for presentation purposes (e.g name, email, image). You can use the session callback to customize the session object returned to the client if you need to return additional data in the session object.

access_token goes under OAuth token.

Check out the jwt and session callbacks: 
https://next-auth.js.org/configuration/callbacks#jwt-callback
https://next-auth.js.org/configuration/callbacks#session-callback

**Reply by harshithpabbati** [reply]
Can we also get accessTokenSecret?

**Reply by balazsorban44** [reply]
so basically it's there, only its called `refreshToken`, I think. see https://github.com/nextauthjs/next-auth/pull/1322#issuecomment-782578376

**Reply by harshithpabbati** [reply]
How can I get that refreshToken for twitter auth? Is it written somewhere, I couldn't find 😓

**Reply by balazsorban44** [reply]
check the account parameter in the jwt callback.
